docs/frontend/preact.md

```diff
@@ -12,3 +12,72 @@ Preact code is located in `app/javascript`.
 
 The components are mounted when needed, look for `javascript_pack_tag` in the
 view pages inside `app/views`.
+
+## PropTypes
+
+Preact supports
+[PropTypes](https://reactjs.org/docs/typechecking-with-proptypes.html). When
+creating Preact components, ensure to always have prop types defined.
+
+### Common PropTypes
+
+PropTypes can become repetitive. In simple cases where a PropType is a string or
+number, this duplication is expected. For commonly used entities in the DEV
+project, common PropTypes exist. All common PropTypes are located in the
+`/app/javascript/src/components/common-prop-types`. For example, the user entity
+that is referenced in many parts of the DEV codebase.
```

Comment:
(I can't make a GitHub suggestion, so I have to do this as a comment)

> PropTypes can become repetitive. In simple cases where a PropType is a string or
number, this duplication is expected. For commonly used entities in the DEV
project, common PropTypes exist. All common PropTypes are located in the
`/app/javascript/src/components/common-prop-types`. For example, the user entity
that is referenced in many parts of the DEV codebase.

Maybe this? _Using PropTypes can be repetitive. Some duplication is normal, like when a PropType is a string or a number. But for commonly-used PropTypes, like the `user` entity, you can use the provided common PropTypes, located in `/app/javascript/src/components/common-prop-types`, as shown below._